packages/frontend-api/src/Model/Resolver/Products/ProductResolverMap.php

```diff
@@ -258,91 +162,40 @@ protected function map(): array
     }
 
     /**
-     * @return array
+     * @return callable[]
      */
     protected function mapProduct(): array
     {
         return [
-            'shortDescription' => function ($data) {
-                return $data instanceof Product ? $data->getShortDescription(
-                    $this->domain->getId()
-                ) : $data['short_description'];
-            },
-            'link' => function ($data) {
-                $productId = $data instanceof Product ? $data->getId() : $data['id'];
-                return $this->getProductLink($productId);
-            },
-            'categories' => function ($data) {
-                return $data instanceof Product ? $data->getCategoriesIndexedByDomainId()[$this->domain->getId()] : $this->getCategoriesForData(
-                    $data
-                );
-            },
-            'flags' => function ($data) {
-                return $this->getFlagsForData($data);
-            },
-            'availability' => function ($data) {
-                return $data instanceof Product ? $data->getCalculatedAvailability() : ['name' => $data['availability']];
-            },
-            'unit' => function ($data) {
-                return $data instanceof Product ? $data->getUnit() : ['name' => $data['unit']];
-            },
-            'stockQuantity' => function ($data) {
-                return $data instanceof Product ? $data->getStockQuantity() : $data['stock_quantity'];
-            },
-            'isUsingStock' => function ($data) {
-                return $data instanceof Product ? $data->isUsingStock() : $data['is_using_stock'];
-            },
-            'brand' => function ($data) {
-                if ($data instanceof Product) {
-                    return $data->getBrand();
-                }
+            self::RESOLVE_FIELD => function ($value, ArgumentInterface $args, ArrayObject $context, ResolveInfo $info) {
+                $methodName = 'get' . ucfirst($info->fieldName);
```

Comment:
I have internall struggle with using only methods prefixed by `get`. E.g. I would like to use `isSellingDenied()` but I cannot without changing this callback.

Could the getting and validating method name be extracted into method?

Reply:
allowed to use `get`, `is`, or no prefix in 01a4cd41fa1b21027952092c8132130c5e95ab78

Reply:
Nice!